voxblox/src/io/mesh_ply.cc

```diff
@@ -51,38 +73,53 @@ bool outputMeshAsPly(const std::string& filename, const Mesh& mesh) {
   stream << "property float x" << std::endl;
   stream << "property float y" << std::endl;
   stream << "property float z" << std::endl;
+  if (mesh.hasNormals()) {
+    stream << "property float normal_x" << std::endl;
+    stream << "property float normal_y" << std::endl;
+    stream << "property float normal_z" << std::endl;
+  }
   if (mesh.hasColors()) {
     stream << "property uchar red" << std::endl;
     stream << "property uchar green" << std::endl;
     stream << "property uchar blue" << std::endl;
+    stream << "property uchar alpha" << std::endl;
```

Comment:
crazy transparent meshes!!!

Reply:
doesn't really work in meshlab or cloud compare, but this way we at least don't lose the info :)

Reply:
Do you know if it works in rviz, I added it as a parameter when I wrote the rviz-plugin but never tested if it does anything

Reply:
Don't know, we didn't really use RVIZ to visualize results, margarita has her own PCL based viewer for GSM and in modelify we dumped data to ply to look at it.